### ISSUE TITLE
cmd/go: 'go test -mod=mod' fails if GOFLAGS contains a default -mod setting

### ISSUE THREAD

**Issue by bcmills** [golang/go#35827]
`go test -mod=mod` does not successfully override a `-mod` flag specified in `GOFLAGS`.
Instead, it fails with `go test: mod flag may be set only once`.

```
go version devel +bb87c192f9 Mon Nov 25 11:43:42 2019 -0500 linux/amd64
```

Script test to reproduce:
```
env GO111MODULE=on
env GOFLAGS=-mod=readonly
go list -mod=mod all
go test -mod=mod .

-- go.mod --
module example.com

go 1.14

require rsc.io/quote latest
-- foo_test.go --
package foo
```

Output:
```
--- FAIL: TestScript (0.01s)
    --- FAIL: TestScript/mod_goflags (0.07s)
        script_test.go:193:
            # An explicit -mod= flag should override one set via GOFLAGS. (0.056s)
            > env GOFLAGS=-mod=readonly
            > go list -mod=mod all
            [stdout]
            example.com
            > go test -mod=mod .
            [stderr]
            go test: mod flag may be set only once
            run "go help test" or "go help testflag" for more information
            [exit status 2]
            FAIL: testdata/script/mod_goflags.txt:7: unexpected command failure
```

CC @jayconrod

**Comment by agnivade** [comment]
I believe this is a duplicate of https://github.com/golang/go/issues/32471.

**Comment by bcmills** [comment]
@agnivade, thanks for the reference. I had overlooked it because `vet` is not implicated in this path.
